WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrderViewModel.swift

```diff
@@ -434,23 +431,14 @@ private extension NewOrderViewModel {
     func configureProductRowViewModels() {
         updateProductsResultsController()
         updateProductVariationsResultsController()
-        productRows = orderSynchronizer.order.items.compactMap { item in
-            guard let productRowViewModel = createProductRowViewModel(for: item, canChangeQuantity: true) else {
-                return nil
+        orderSynchronizer.orderPublisher
+            .map { $0.items }
+            .removeDuplicates()
```

Comment:
What's the purpose of removing duplicates here?

The only case I can think of is that this will remove a duplicate item, if the same item is added more than once to an order. But up to now we have intentionally allowed this — Core also allows multiple line items for the same product, and we have a separate issue (178-gh-woocommerce/woomobile-private) for possibly changing that behavior.

Reply:
Hey @rachelmcr, this won't remove items with the same product because those items have a different `id`(randomly generated). This is "needed" to not redraw all items every time something else from the order changes, like shipping, fees, etc.

Does that makes sense?

Reply:
Ah, got it! Thanks for explaining.